src/app/beer_garden/plugin.py

```diff
@@ -302,6 +305,76 @@ def read_logs(
     return request
 
 
+@publish_event_async(Events.INSTANCE_UPDATED)
+async def update_async(
+    instance_id: str = None,
+    instance: Instance = None,
+    system: System = None,
+    new_status: str = None,
+    metadata: dict = None,
+    **_,
+) -> dict:
+    """Update an Instance status.
+
+    Will also update the status_info heartbeat.
+
+    Args:
+        instance_id: The Instance ID
+        instance: The Instance
+        system: The System
+        new_status: The new status
+        metadata: New metadata
+
+    Returns:
+        The updated Instance
+    """
+    query = {"instances._id": ObjectId(instance_id)}
+    projection = {"instances.$": 1, "_id": 0}
+    update = {}
```

Comment:
This is kind of an overarching comment on this file. I get it that we need to create these objects, but in my ideal world this stuff would be pushed down to the DB code some home. Because we are doing this for a select few of the commands, yeah we can't do it. But I still want us to think about how we could handle that going forward. If everything migrated over to motor, then I would not want to see this here.

Reply:
Completely agree. I think the correct thing to do is to start to make use of our db "api". That's going to be involved... I'll make an issue to track it.

Reply:
#634